Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -129,11 +136,91 @@ public void RespectAnnotationInheritance()
         public void RespectAnnotationPrefixes()
         {
             AssertHasAnnotation(
-                "my.prefix.Truck", 
+                "my.prefix.Truck",
                 new IonSerializer(new IonSerializationOptions { IncludeTypeInformation = true, TypeAnnotationPrefix = new FixedTypeAnnotationPrefix("my.prefix") }).Serialize(new Truck()));
             AssertHasAnnotation(
-                "my.universal.namespace.BussyMcBusface", 
+                "my.universal.namespace.BussyMcBusface",
                 new IonSerializer().Serialize(new Bus()));
         }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToParentClassNameMatchingAnnotation()
+        {
+            IIonValue ionTruck = valueFactory.NewEmptyStruct();
+            ionTruck.AddTypeAnnotation("Truck");
+
+            IIonReader reader = IonReaderBuilder.Build(ionTruck);
+
+            IonSerializer ionSerializer = new IonSerializer();
+            Vehicle truck = ionSerializer.Deserialize<Vehicle>(reader);
+
+            Assert.AreEqual(truck.ToString(), nativeTruck.ToString());
+        }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToClassNameMatchingAnnotation()
+        {
+            IIonValue ionTruck = valueFactory.NewEmptyStruct();
+            ionTruck.AddTypeAnnotation("Truck");
+
+            IIonReader reader = IonReaderBuilder.Build(ionTruck);
+
+            IonSerializer ionSerializer = new IonSerializer();
+            Truck truck = ionSerializer.Deserialize<Truck>(reader);
+
+            Assert.AreEqual(truck.ToString(), nativeTruck.ToString());
+        }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToClassNameNotMatchingAnnotation()
+        {
+            IIonValue ionTruck = valueFactory.NewEmptyStruct();
+            ionTruck.AddTypeAnnotation("NonMatching");
```

Comment:
So this "NonMatching" annotation makes no difference to the behaviour compared to the "Truck" annotation (from the previous test)?

Reply:
The test is to ensure that if the annotation doesn't match anything, it takes the passed in type as what it should be mapping to.